File: shared/src/main/java/ru/privatenull/pnauth/api/AuthAdmissionApi.java
package ru.privatenull.pnauth.api;

import java.util.concurrent.CompletableFuture;

/** Admission checks used by proxy adapters before routing a player. */
public interface AuthAdmissionApi {
    CompletableFuture<AdmissionDecision> checkAdmission(String username, String ip, int onlineAccountsFromIp);
    CompletableFuture<Boolean> isPremium(String username);
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/api/AuthAdmissionApi.kt
package ru.privatenull.pnauth.api

import java.util.concurrent.CompletableFuture

/** Admission checks used by proxy adapters before routing a player. */
interface AuthAdmissionApi {
    fun checkAdmission(username: String, ip: String, onlineAccountsFromIp: Int): CompletableFuture<AdmissionDecision>
    fun isPremium(username: String): CompletableFuture<Boolean>
}
